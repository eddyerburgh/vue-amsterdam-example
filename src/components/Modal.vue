<template>
  <div v-if="visible" class="modal is-active">
    <div class="modal-background"></div>
    <div class="modal-contents">
      <div class="box">
        <button
          @click="onClose"
          class="delete"
          aria-label="close">
        </button>
        <slot />
        </div>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  props: ['visible', 'onClose']
}
</script>
