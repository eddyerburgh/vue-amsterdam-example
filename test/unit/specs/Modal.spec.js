import Modal from '../../../src/components/Modal.vue'
import { mount } from '@vue/test-utils'

test('does not render when not passed visible prop', () => {
  const wrapper = mount(Modal)
  expect(wrapper.isEmpty()).toBe(true)
})

test('renders when passed visible prop as true', () => {
  const wrapper = mount(Modal, {
    propsData: {
      visible: true
    }
  })
  expect(wrapper.isEmpty()).toBe(false)
})

test('calls onClose when button is clicked', () => {
  const onClose = jest.fn()
  const wrapper = mount(Modal, {
    propsData: {
      visible: true,
      onClose
    }
  })
  wrapper.find('button').trigger('click')
  expect(onClose).toHaveBeenCalled()
})

test('renders correctly', () => {
  const wrapper = mount(Modal, {
    propsData: {
      visible: true
    },
    slots: {
      default: '<p>some content</p>'
    }
  })
  expect(wrapper.html()).toMatchSnapshot()
})
