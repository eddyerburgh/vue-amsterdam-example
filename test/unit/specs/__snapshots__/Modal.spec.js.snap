// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div class="modal is-active">
    <div class="modal-background"></div>
    <div class="modal-contents">
        <div class="box">
            <button aria-label="close" class="delete"></button>
            <p>some content</p>
        </div>
    </div>
</div>
`;
